### ISSUE TITLE
output ghc warnings for multi-package build

### ISSUE THREAD

**Issue by cdepillabout** [commercialhaskell/stack#896]
When running `cabal build` or `stack build` from the command line, I like to see the warnings produced by ghc.  This works well when calling `stack build` from a single-package project.  As `stack build` is running, it outputs the ghc warnings for each file it is compiling.

However, for a multi-package build, `stack build` doesn't output warnings from ghc.

Is there a flag to `stack build` that will output the ghc warnings?

It is possible [this issue](https://github.com/commercialhaskell/stack/issues/426) is related.  However, I don't want to see ghc warnings for ALL dependencies, just for the packages in my `stack.yaml` file.

**Comment by mgsloan** [comment]
See also the discussion in https://github.com/commercialhaskell/stack/issues/298

**Comment by borsboom** [comment]
Have you tried `stack build --pedantic`?  It doesn't do exactly what you want, but since it enables `-Werror` it means any warnings will cause your build to fail, and so you won't miss any.  If you were previously building without it, it will also trigger a rebuild of all the modules, so you won't miss warnings for modules that had previous been built (but that you ignored the warnings for).  I almost always use `--pedantic`.

Overall, showing warnings only seems kind of complicated.  Filtering GHC output that's meant for human consumption this way would be error-prone, and we'd have to deal with not interlacing output when multiple packages are building in parallel.  I'd be more in favour of getting #426 done first, perhaps making it more advanced by being able to specify that only local packages should get full output.

**Comment by cdepillabout** [comment]
I have been using `--pedantic` as a work around, but there are lots of times I don't want to build to fail when there are warnings (but I still want to see the warnings).

> Filtering GHC output that's meant for human consumption this way would be error-prone, and we'd have to deal with not interlacing output when multiple packages are building in parallel.

I would be okay with seeing all of the ghc output (since the warnings are in there as well).  Is there some way to print all of the ghc output to the console?

**Comment by borsboom** [comment]
That's what #426 is about.  Perhaps add a comment there about an option to only print GHC output for local packages, and close this issue in its favour?

**Comment by cdepillabout** [comment]
Okay, sounds good.

Closing in favor of #426.

**Comment by snoyberg** [comment]
Relevant PR: https://github.com/commercialhaskell/stack/pull/4097
